src/Opdex.Platform.Domain/Models/Tokens/TokenChain.cs

```diff
@@ -0,0 +1,30 @@
+using Opdex.Platform.Common.Extensions;
+using System;
+
+namespace Opdex.Platform.Domain.Models.Tokens;
+
+public class TokenChain
+{
+    public TokenChain(ulong tokenId, ExternalChainType nativeChain, string nativeAddress)
+    {
+        if (tokenId == 0) throw new ArgumentOutOfRangeException(nameof(tokenId), "Token id must be greater than 0.");
+        if (!nativeChain.IsValid()) throw new ArgumentOutOfRangeException(nameof(nativeChain), "Chain type must be valid.");
+        if (!nativeAddress.HasValue()) throw new ArgumentNullException(nameof(nativeAddress), "Native address must be set.");
```

Comment:
In the case that Bitcoin or a blockcore chain is ever supported w/ out smart contracts, native address should be allowed to be null.

Reply:
Good spot. This is an oversight, my intention was that for the base token of the chain, the `NativeAddress` would be empty/null